src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -94,8 +118,49 @@ public VisualStudioProjectTracker(IServiceProvider serviceProvider)
             }
         }
 
+        private void WaitForScheduledForegroundAffinitizedActions()
+        {
+            AssertIsForeground();
+
+            lock (_gate)
+            {
+                _taskForForegroundAffinitizedActions.Wait();
```

Comment:
I haven't tested it, most probably it doesn't. I am actually not even sure we need to wait for all scheduled foreground actions here: https://github.com/dotnet/roslyn/pull/12927/files/55b4181c4040cae49c209fc23dd9b8e829cefb55#diff-5561564909d81f97c70a52c751280162R225. These scheduled tasks may add/remove projects, but that shouldn't be no different then these events coming after we have started pushing to workspace host. I am tending towards just removing this, what do you suggest?